subprojects/logging/src/main/java/org/gradle/internal/logging/sink/OutputEventRenderer.java

```diff
@@ -329,6 +352,10 @@ public void onOutput(OutputEvent event) {
                 return;
             }
             this.maxWorkerCount.set(newMaxWorkerCount);
+        } else if (event instanceof EndOutputEvent) {
+            if (updateNowExecutor != null) {
+                updateNowExecutor.shutdownNow();
```

Comment:
It had no effect on the regression and if not used carefully can drop events. I wouldn't do it.